.circleci/config.yml

```diff
@@ -101,125 +173,185 @@ jobs:
       - run: bazel test @program_example//...
       - run: bazel test @packages_example//...
 
-  # This job cannot run on BuildKite (BazelCI) because it requires changing
-  # directories to test inside nested workspaces.
-  # See https://github.com/bazelbuild/continuous-integration/issues/225
-  test_e2e_legacy:
+  build_release:
     <<: *job_defaults
-    environment:
-      CHROMEDRIVER_VERSION_ARG: --versions.chrome 2.41
     steps:
-      - checkout:
-          <<: *post_checkout
-      - *setup_bazelrc
-
-      # Not using "bazel run @nodejs//:yarn test" as will attempt to run
-      # the "test" script on all folder listed in node_repositories()
-      # These e2e tests are in the process of being moved to /e2e
-      - run: yarn test:e2e
+      - *attach_workspace
+      - *init_environment
+      - *init_bazel
+      - *hide_node_and_yarn_local_binaries
+      - run: ./scripts/build_release.sh
+      - persist_to_workspace:
+          root: ~/
+          paths:
+            - ./rules_nodejs/dist/build_bazel_rules_nodejs
 
-  # This job cannot run on BuildKite (BazelCI) because it requires changing
-  # directories to test inside nested workspaces.
-  # See https://github.com/bazelbuild/continuous-integration/issues/225
-  test_e2e:
+  build_typescript_package:
     <<: *job_defaults
+    # Need to compile Go and protocol buffers from source for typescript package
     resource_class: xlarge
-    parallelism: 4
-    environment:
-      CHROMEDRIVER_VERSION_ARG: --versions.chrome 2.41
     steps:
-      - checkout:
-          <<: *post_checkout
-      - *setup_bazelrc
+      - *attach_workspace
+      - *init_environment
+      - *init_bazel
+      - *hide_node_and_yarn_local_binaries
+      - run: ./scripts/build_packages.sh typescript
+      - persist_to_workspace:
+          root: ~/
+          paths:
+            - ./rules_nodejs/dist/npm_bazel_typescript
 
-      # Some e2e tests such as ts_devserver require xvfb
-      - run: xvfb-run -a ./e2e/test.sh ${CIRCLE_NODE_INDEX} ${CIRCLE_NODE_TOTAL}
+  build_karma_package:
+    <<: *job_defaults
+    steps:
+      - *attach_workspace
+      - *init_environment
+      - *init_bazel
+      - *hide_node_and_yarn_local_binaries
+      - run: ./scripts/build_packages.sh karma
+      - persist_to_workspace:
+          root: ~/
+          paths:
+            - ./rules_nodejs/dist/npm_bazel_karma
 
-  # This job cannot run on BuildKite (BazelCI) because it requires changing
-  # directories to test inside nested workspaces.
-  # See https://github.com/bazelbuild/continuous-integration/issues/225
-  test_examples:
+  build_jasmine_package:
+    <<: *job_defaults
+    steps:
+      - *attach_workspace
+      - *init_environment
+      - *init_bazel
+      - *hide_node_and_yarn_local_binaries
+      - run: ./scripts/build_packages.sh jasmine
+      - persist_to_workspace:
+          root: ~/
+          paths:
+            - ./rules_nodejs/dist/npm_bazel_jasmine
+
+  build_labs_package:
```

Comment:
next PR: I'd like to reduce number of steps to create a new package, maybe we can build all of these in one job in a loop?

Reply:
yup. that would be ideal. we could shard it like that. the only complication and why I didn't in this PR was that the karma package depends on typescript package being build first.